module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -808,33 +808,35 @@ public function placeHold($holdDetails)
         } catch (Exception $e) {
             throw new ILSException('hold_date_invalid');
         }
-        // Get status to separate Holds (on checked out items) and Pages on a
-        $status = $this->getStatus($holdDetails['item_id']);
         $requestBody = [
-            'requestType' => $item->status->name == 'Available' ? 'Page' : 'Hold',
-            'requestDate' => date('c'),
-            'requesterId' => $holdDetails['patron']['id'],
-            'requester' => [
-                'firstName' => $holdDetails['patron']['firstname'] ?? '',
-                'lastName' => $holdDetails['patron']['lastname'] ?? ''
-            ],
             'itemId' => $holdDetails['item_id'],
+            'requestType' => $holdDetails['status'] == 'Available'
+                ? 'Page' : 'Recall',
+            'requesterId' => $holdDetails['patron']['id'],
+            'requestDate' => date('c'),
             'fulfilmentPreference' => 'Hold Shelf',
             'requestExpirationDate' => date_format($requiredBy, 'Y-m-d'),
+            'pickupServicePointId' => $holdDetails['pickUpLocation']
         ];
         $response = $this->makeRequest(
             'POST',
-            '/request-storage/requests',
+            '/circulation/requests',
             json_encode($requestBody)
         );
         if ($response->isSuccess()) {
-            return [
+            $json = json_decode($response->getBody());
+            $result = [
                 'success' => true,
-                'status' => $response->getBody()
+                'status' => $json->status
             ];
         } else {
-            throw new ILSException($response->getBody());
+            $json = json_decode($response->getBody());
```

Comment:
It would probably be a good idea to account for the possibility that the response is not JSON here, in case of a more generalized server failure.

Reply:
Thanks, forgive my basic php, but I'm thinking something like this
```
            try {
                $json = json_decode($response->getBody());
                $result = [
                    'success' => false,
                    'status' => $json->errors[0]->message
                ];
            } catch (Exception $e) {
                throw new ILSException($response->getBody());
            }
```
Would it be better still to not to assume we can get a response and just throw `$e->getMessage()`?

Reply:
I think what you propose makes sense. I suppose we could concatenate both $e->getMessage and the body together for the most comprehensive possible exception message, but without a real-world example, it's hard to know whether that would be better or worse. We can start here and revise as experience informs us to. :-)